marshal_test.go

```diff
@@ -162,3 +163,43 @@ func TestDate_UnmarshalText_invalid_date_text(t *testing.T) {
 		}
 	}
 }
+
+func TestDate_UnmarshalText_empty(t *testing.T) {
+	cases := []struct {
+		value    string
+		expected Date
+	}{
+		{``, New(1, time.January, 1)},
```

Comment:
this passes, proving that empty string parses as the date zero value - I believe that in v1 the zero value was epoch, which is where we saw this initially

Reply:
Both v1 and v2 have the same UnmarshalText behaviour. The caveat is that in v1, the zero value was 1st Jan 1970; in v2 this has moved back to 1st Jan 1AD (the same as time.Time uses).

`UnmarshalText` states "... that the a blank string is unmarshalled as the zero value." Are you hoping for this to change?

Reply:
Thanks for the quick review!

I was hoping for the behavior to change. I had found it surprising that a `Date` in a struct would convert an empty string to a zero value date using the built-in unmarshal function

Also apologies that one of the tests I added does fail - I had added it to showcase what I believed was unintended behavior: https://github.com/rickb777/date/pull/24#discussion_r1691543911